### ISSUE TITLE
About the visualization of pick and place affordance

### ISSUE THREAD

**Issue by qianLyu** [cliport/cliport#19]
Dear author,

Thanks for your code about heatmap visualization! Following [https://github.com/cliport/cliport/blob/master/notebooks/affordances.ipynb](url), I can successfully visualize the pick and place location in my own dataset. However, what I see in the heatmap are just some tiny bright points (maybe scale a few pixels) indicating the target location, which are quite different from your figures (the high-lightened areas exactly match the shape of target block). I'm wondering the reason. From my understanding, since the target position is a single pixel, the output logits after successful training should be concentrated on the target pixel. So that the high-lightened area should be focused on certain point(s). Do you have any idea about this?

Thanks!

**Comment by MohitShridhar** [comment]
@qianLyu, sorry, I am a bit confused. The affordances in Fig 4 and Fig 10 were generated with the same affordances.ipynb notebook. Can you provide a concrete example of what is different?

**Comment by qianLyu** [comment]
@MohitShridhar, thanks for your reply! My question is about the size of the 'affordance prediction area'. In Fig 4 sim setting, the affordance area of picking contains the whole block. In Fig 4 real setting however, the affordance area of picking just contains tiny bright points. I think the latter one does make sense, since the labels are single pixels, and that the output prediction should match the pixel. What confuses me is just the size of picking affordance in sim setting.

**Comment by MohitShridhar** [comment]
@qianLyu, I think I see what you mean. Why is the sim pick distribution more "spread out" compared to the real  distribution which is "peaky"? It's probably to do with the amount of training data. In sim, the agent seen 1000s of examples where picking anywhere on the block leads to successes. Whereas in the real-world, the agent has only seen a handful number of pick actions, and hence overfits to a specific location, resulting in the peaky distribution. I think if you train the real-world agent with lots of diverse training data, it will probably resemble the sim affordances.

**Comment by qianLyu** [comment]
@MohitShridhar, got it! It's reasonable. Thanks a lot.
